dictdiffer/utils.py

```diff
@@ -264,10 +264,19 @@ def are_different(first, second, tolerance):
         return False
 
     first_is_nan, second_is_nan = bool(first != first), bool(second != second)
+    first_is_inf, second_is_inf = first == float('inf'), second == float('inf')
+    first_is_ninf = first == float('-inf')
+    second_is_ninf = second == float('-inf')
 
     if first_is_nan or second_is_nan:
```

Comment:
on a more mathematical level, should two `nan` or two `inf` or two `-inf` be considered the same? It is improbable that two `inf` are the same when you are running a diff. If we agree that they should be considered the same, then this all looks great!

Reply:
I asked myself the same question while doing this PR and I found links like [this](https://stackoverflow.com/questions/41834621/c-ieee-floats-inf-equal-inf) stating that `inf==inf` and `-inf==-inf` .

On the contrary, it seems that they agree that `Nan!=Nan` , but  that's outside the scope of this PR.

Let me know if you prefer not considering two `inf` the same, I can easily adapt tests and code.

Reply:
I think following convention is preferred over mathematical precision, so this sounds good to me 👍